File: debug_test_send.py
#!/usr/bin/env python3
"""
Debug Script: Testar envio de mensagens via Edge Function
Captura o erro real do fluxo: Frontend → Edge Function → Z-API

Uso:
  python debug_test_send.py
"""

import requests
import json
import os
from dotenv import load_dotenv

# Carregar variáveis de ambiente
load_dotenv()

SUPABASE_URL = os.getenv("VITE_SUPABASE_URL") or input("VITE_SUPABASE_URL: ")
SUPABASE_ANON_KEY = os.getenv("VITE_SUPABASE_ANON_KEY") or input("VITE_SUPABASE_ANON_KEY: ")
TEST_PHONE = input("Número WhatsApp para teste (ex: 85 988889999): ")

print("\n" + "="*70)
print("🧪 DEBUG: Teste de Envio de Mensagem WhatsApp")
print("="*70)

# 1. Informações da requisição
edge_function_url = f"{SUPABASE_URL}/functions/v1/enviar-whatsapp"
headers = {
    "Content-Type": "application/json",
    "Authorization": f"Bearer {SUPABASE_ANON_KEY}",
}
payload = {
    "telefone": TEST_PHONE,
    "evento": "solicitacao.criada",
    "variaveis": {
        "cliente_nome": "Teste Debug",
        "solicitacao_id": "DEBUG-001",
        "endereco_coleta": "Rua de Teste, 123",
        "endereco_entrega": "Avenida Teste, 456",
        "valor_total": "R$ 100,00",
    },
}

print("\n📤 REQUISIÇÃO")
print(f"URL: {edge_function_url}")
print(f"Método: POST")
print(f"Headers: {json.dumps(headers, indent=2)}")
print(f"Body: {json.dumps(payload, indent=2)}")

# 2. Fazer requisição
print("\n⏳ Enviando requisição...")
try:
    response = requests.post(edge_function_url, json=payload, headers=headers, timeout=30)
    
    print("\n📡 RESPOSTA")
    print(f"Status: {response.status_code} {response.reason}")
    print(f"Headers: {dict(response.headers)}")
    
    try:
        data = response.json()
        print(f"Body: {json.dumps(data, indent=2, ensure_ascii=False)}")
    except:
        print(f"Body (texto): {response.text}")
    
    # 3. Análise
    print("\n🔍 ANÁLISE")
    if response.status_code == 200:
        print("✅ Edge Function respondeu com sucesso!")
        data = response.json()
        if data.get("success"):
            print("   ✅ Mensagem foi enviada para Z-API!")
            print(f"   📨 ID da mensagem: {data.get('zapiMessageId')}")
        else:
            print(f"   ❌ Erro no envio: {data.get('error')}")
            if "Z-API" in str(data.get('error')):
                print("   → Problema com Z-API (credenciais, configuração ou servidor)")
    
    elif response.status_code == 422:
        print("⚠️  Configuração incompleta (422)")
        data = response.json()
        print(f"   • Erro: {data.get('error')}")
        print(f"   • Código: {data.get('errorCode')}")
        if data.get('details'):
            print(f"   • Detalhes: {data.get('details')}")
    
    elif response.status_code == 502:
        print("❌ Erro 502 - Problema ao conectar com Z-API")
        data = response.json()
        print(f"   • Erro: {data.get('error')}")
        print(f"   • Status Z-API: {data.get('zapiStatus')}")
        print(f"   • Resposta Z-API: {data.get('zapiResponse')}")
    
    elif response.status_code == 401:
        print("❌ Erro 401 - Autenticação falhou")
        print("   → Verifique se o VITE_SUPABASE_ANON_KEY está correto")
    
    elif response.status_code == 404:
        print("❌ Erro 404 - Edge Function não encontrada")
        print(f"   → URL: {edge_function_url}")
    
    else:
        print(f"❌ Erro {response.status_code}: {response.reason}")
    
    # 4. Recomendações
    print("\n💡 PRÓXIMOS PASSOS")
    
    if response.status_code == 422:
        print("1. Verifique as Integrações em Configurações")
        print("   • Integração 'WhatsApp via Z-API' deve estar ativa")
        print("   • instance_id e token devem estar preenchidos")
    
    elif "INCOMPLETE_CREDENTIALS" in str(response.json().get('errorCode')):
        print("1. Configure a integração Z-API em Configurações > Integrações")
        print("2. Preencha instance_id, token e client_token")
    
    elif response.status_code == 502:
        print("1. Verifique se as credenciais Z-API estão corretas")
        print("2. Teste a Z-API diretamente em: https://dashboard.z-api.io")
        print("3. Verifique se a instância WhatsApp está ativa")
    
    else:
        print("1. Verifique os logs da Edge Function em Supabase Dashboard")
        print("2. Procure por mensagens [enviar-whatsapp] nos logs")

except requests.exceptions.Timeout:
    print("❌ TIMEOUT - Requisição demorou muito")
    print("   → Edge Function demora > 30s para responder")

except requests.exceptions.ConnectionError as e:
    print(f"❌ ERRO DE CONEXÃO: {e}")
    print(f"   → Verifique URL: {edge_function_url}")

except Exception as e:
    print(f"❌ ERRO: {e}")

print("\n" + "="*70)

File: debug_test_send.js
/**
 * Debug Script: Teste de Conexão com Edge Function
 * Simula o que o frontend faz quando clica "Enviar Teste"
 * 
 * Uso:
 *   npm run debug:testsend
 *   ou
 *   node debug_test_send.js
 */

import fetch from 'node-fetch';

// Configurar variáveis
const SUPABASE_URL = process.env.VITE_SUPABASE_URL || 'https://seu-projeto.supabase.co';
const SUPABASE_ANON_KEY = process.env.VITE_SUPABASE_ANON_KEY || 'sua-chave-anon';
const TEST_PHONE = process.argv[2] || '85988889999';

console.log('\n' + '='.repeat(70));
console.log('🧪 DEBUG: Teste de Conexão com Edge Function');
console.log('='.repeat(70));

// URL da Edge Function
const url = `${SUPABASE_URL}/functions/v1/enviar-whatsapp`;

// Payload
const payload = {
  telefone: TEST_PHONE,
  evento: 'solicitacao.criada',
  variaveis: {
    cliente_nome: 'Cliente Teste Debug',
    solicitacao_id: 'DEBUG-001',
    endereco_coleta: 'Rua de Teste, 123',
    endereco_entrega: 'Avenida Teste, 456',
    valor_total: 'R$ 100,00',
  },
};

console.log('\n📤 REQUISIÇÃO');
console.log(`URL: ${url}`);
console.log(`Método: POST`);
console.log(`Headers: {
  "Content-Type": "application/json",
  "Authorization": "Bearer ${SUPABASE_ANON_KEY.substring(0, 10)}..."
}`);
console.log(`Body: ${JSON.stringify(payload, null, 2)}`);

// Fazer requisição
console.log('\n⏳ Enviando requisição...');

fetch(url, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${SUPABASE_ANON_KEY}`,
  },
  body: JSON.stringify(payload),
})
  .then(response => {
    console.log('\n📡 RESPOSTA');
    console.log(`Status: ${response.status} ${response.statusText}`);
    console.log(`Headers:`, {
      contentType: response.headers.get('content-type'),
      cors: response.headers.get('access-control-allow-origin'),
    });

    return response.json().then(data => ({ response, data }));
  })
  .then(({ response, data }) => {
    console.log(`Body: ${JSON.stringify(data, null, 2)}`);

    console.log('\n🔍 ANÁLISE');

    // Análise por status
    if (response.status === 200) {
      console.log('✅ Edge Function respondeu com sucesso!');

      if (data.success) {
        console.log('   ✅ Mensagem foi enviada para Z-API!');
        console.log(`   📨 ID da mensagem: ${data.zapiMessageId}`);
      } else {
        console.log(`   ❌ Erro no envio: ${data.error}`);
        if (data.zapiStatus) {
          console.log(`   📡 Status Z-API: ${data.zapiStatus}`);
          console.log(`   📦 Resposta Z-API:`, data.zapiResponse);
        }
      }
    } else if (response.status === 422) {
      console.log('⚠️  Configuração incompleta (422)');
      console.log(`   • Erro: ${data.error}`);
      console.log(`   • Código: ${data.errorCode}`);
      if (data.details) {
        console.log(`   • Detalhes:`, data.details);
      }

      console.log('\n💡 AÇÃO NECESSÁRIA');
      if (data.errorCode === 'INTEGRATION_NOT_FOUND') {
        console.log('1. Vá para Configurações > Integrações');
        console.log('2. Configure "WhatsApp via Z-API"');
        console.log('3. Preencha instance_id, token e client_token');
      }
    } else if (response.status === 502) {
      console.log('❌ Erro 502 - Problema ao conectar com Z-API');
      console.log(`   • Erro: ${data.error}`);
      console.log(`   • Status Z-API: ${data.zapiStatus}`);
      console.log(`   • Resposta Z-API:`, data.zapiResponse);

      console.log('\n💡 AÇÃO NECESSÁRIA');
      console.log('1. Verifique se as credenciais Z-API estão corretas');
      console.log('2. Teste em: https://dashboard.z-api.io');
      console.log('3. Verifique se a instância WhatsApp está ativa');
    } else if (response.status === 401) {
      console.log('❌ Erro 401 - Autenticação falhou');
      console.log('   → VITE_SUPABASE_ANON_KEY pode estar incorreta');
    } else if (response.status === 404) {
      console.log('❌ Erro 404 - Edge Function não encontrada');
      console.log(`   → URL: ${url}`);
    } else {
      console.log(`❌ Erro ${response.status}: ${response.statusText}`);
    }

    console.log('\n' + '='.repeat(70));
  })
  .catch(error => {
    console.error('\n❌ ERRO:', error.message);
    console.error('\n💡 VERIFICAÇÕES');
    console.error('• VITE_SUPABASE_URL está correto?');
    console.error('• VITE_SUPABASE_ANON_KEY está correto?');
    console.error('• Sua internet está funcionando?');
    console.error('• Edge Function foi deployada?');
    console.log('\n' + '='.repeat(70));
  });
